src/simulation.hpp

```diff
@@ -901,14 +900,18 @@ template <typename problem_t> void AMRSimulation<problem_t>::ellipticSolveAllLev
 
 		// solve Poisson equation with open b.c. using the method of James (1977)
 		amrex::Vector<amrex::MultiFab> phi(finest_level + 1);
+		// define another fab to store phi
+		amrex::Vector<amrex::MultiFab> phi_data(finest_level + 1);
 		amrex::Vector<amrex::MultiFab> rhs(finest_level + 1);
 		const int nghost = 1;
 		const int ncomp = 1;
 		amrex::Real rhs_min = std::numeric_limits<amrex::Real>::max();
 		for (int lev = 0; lev <= finest_level; ++lev) {
 			phi[lev].define(grids[lev], dmap[lev], ncomp, nghost);
```

Comment:
You could just get rid of `phi` and use `phi_data` instead. If the potential is kept in memory anyway, there's no point in creating a separate variable and copying it.